cli/internal/terraform/terraform/azure/main.tf

```
@@ -0,0 +1,195 @@
+  loadbalancer_id = azurerm_lb.loadbalancer.id
+  name            = "${var.name}-all"
+}
+
+resource "azurerm_lb_outbound_rule" "outbound" {
+  name                    = "${var.name}-outbound"
+  loadbalancer_id         = azurerm_lb.loadbalancer.id
+  protocol                = "All"
+  backend_address_pool_id = azurerm_lb_backend_address_pool.all.id
+
+  frontend_ip_configuration { name = "PublicIPAddress" }
+}
+
+resource "azurerm_virtual_network" "network" {
+  name                = local.name
+  resource_group_name = var.resource_group
+  location            = var.location
+  address_space       = ["10.0.0.0/8"]
+  tags                = local.tags
+}
+
+resource "azurerm_subnet" "node_subnet" {
+  name                 = "${local.name}-node"
+  resource_group_name  = var.resource_group
+  virtual_network_name = azurerm_virtual_network.network.name
+  address_prefixes     = ["10.9.0.0/16"]
+}
+
+resource "azurerm_subnet" "pod_subnet" {
+  name                 = "${local.name}-pod"
+  resource_group_name  = var.resource_group
+  virtual_network_name = azurerm_virtual_network.network.name
+  address_prefixes     = ["10.10.0.0/16"]
+}
+
+resource "azurerm_network_security_group" "security_group" {
+  name                = local.name
+  location            = var.location
+  resource_group_name = var.resource_group
+  tags                = local.tags
+
+  dynamic "security_rule" {
+    for_each = flatten([
+      { name = "noderange", priority = 100, dest_port_range = local.ports_node_range },
+      { name = "kubernetes", priority = 101, dest_port_range = local.ports_kubernetes },
+      { name = "bootstrapper", priority = 102, dest_port_range = local.ports_bootstrapper },
+      { name = "konnectivity", priority = 103, dest_port_range = local.ports_konnectivity },
+      { name = "recovery", priority = 104, dest_port_range = local.ports_recovery },
+      var.debug ? [{ name = "debugd", priority = 105, dest_port_range = local.ports_debugd }] : [],
+    ])
+    content {
+      name                       = security_rule.value.name
+      priority                   = security_rule.value.priority
+      direction                  = "Inbound"
+      access                     = "Allow"
+      protocol                   = "Tcp"
+      source_port_range          = "*"
+      destination_port_range     = security_rule.value.dest_port_range
+      source_address_prefix      = "*"
+      destination_address_prefix = "*"
+    }
+  }
+}
+
+module "scale_set_control_plane" {
+  source = "./modules/scale_set"
+
+  name                      = "${local.name}-control-plane"
+  instance_count            = var.control_plane_count
+  state_disk_size           = var.state_disk_size
+  state_disk_type           = var.state_disk_type
+  resource_group            = var.resource_group
+  location                  = var.location
+  instance_type             = var.instance_type
+  tags                      = merge(local.tags, { role = "control-plane" })
```

```suggestion
  tags                      = merge(local.tags, { constellation-role = "control-plane" })
```
AWS and GCP use the `constellation-role` tag/label
This will most likely require changes to the metadata API

---

Can you fix this in #242 ? I don't want to change the metadata API again in this PR, and the tag was 'uid' on Azure before.